modules/cloudsql-postgres/main.tf

```diff
@@ -0,0 +1,122 @@
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+    }
+  }
+}
+
+# Labels
+
+locals {
+  default_labels = {
+    cloudsql = var.name
+  }
+
+  squad_label = var.squad != "" ? {
+    squad = var.squad
+    team  = var.squad
+  } : {}
+
+  merged_labels = merge(local.default_labels, local.squad_label)
+}
+
+# Primary Cloud SQL instance
+
+resource "google_sql_database_instance" "this" {
+  name    = var.name
+  project = var.project
+  region  = var.region
+
+  database_version    = var.database_version
+  deletion_protection = var.deletion_protection
+
+  settings {
+    tier                  = var.tier
+    availability_type     = var.enable_high_availability ? "REGIONAL" : "ZONAL"
+    disk_type             = "PD_SSD"
+    disk_size             = var.storage_gb
+    disk_autoresize       = true
+    disk_autoresize_limit = var.disk_autoresize_limit
```

Comment:
if we do `disk_`autoresize = true`, i dont think we want to give user the option to specify disk size

```
Note that this value will override the resizing from disk_autoresize if that feature is enabled. To avoid this, set lifecycle.ignore_changes on this field.
```
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#disk_size-1